### PR TITLE
add switch case by kind functions

### PR DESCRIPTION
**Pull request by vincascm**

### CODE REVIEW

**Comment by A-Walrus** [comment]
LGTM, although you need to run `cargo xtask docgen` to update the auto-generated documentation to include the new commands

**Comment by vincascm** [comment]
thank you for your reply. this is just a set of `MappableCommand::Static`, even if `cargo xtask docgen` executed, will not change `book/src/generated/typable-cmd.md`

File: helix-term/Cargo.toml [hunk 1]
```diff
@@ -67,6 +67,9 @@ serde = { version = "1.0", features = ["derive"] }
 grep-regex = "0.1.10"
 grep-searcher = "0.1.10"
 
+# case conversion
+heck = "0.4.0"
```
Comment (archseer) [inline helix-term/Cargo.toml:71]: Pulls in another dependency :/

**Comment by the-mikedavis** [comment]
I don't think it's necessary to add commands for this. Converting case is possible to do with just regular editing primitives. The Kakoune introduction docs even use it to show off the power of selections and multiple cursors: https://kakoune.org/why-kakoune/why-kakoune.html#_multiple_selections

File: helix-term/Cargo.toml (hunk 1 above)
Comment (vincascm) [reply helix-term/Cargo.toml:71]: `heck` has only one dependency : [unicode-segmentation = { version = "1.2.0", optional = true }](https://github.com/withoutboats/heck/blob/dbcfc7b8db8e532d1fad44518cf73e88d5212161/Cargo.toml#L20)

**Comment by vincascm** [comment]
@the-mikedavis consider any conversion between the following [7 cases](https://github.com/withoutboats/heck#cases-contained-in-this-library), if regular editing primitives is used, it is not easy to do.
but these operations can indeed be composed of regular editing primitives, probably lacking a good mapping, between key bindings and a sequence of operations.

File: helix-term/Cargo.toml (hunk 1 above)
Comment (archseer) [reply helix-term/Cargo.toml:71]: Yeah but heck is a new dependency that's pulled in for just a couple commands.

**Comment by archseer** [comment]
If we added macro keybinds then this would be easier to map to a key. A simpler implementation that doesn't pull in heck could also just set the macro register to a sequence of these keys, then execute the macro

**Comment by vincascm** [comment]
ok, thanks for your replies

**Comment by peterwilli** [comment]
I made my own solution after not finding anything satisfactory... 

See demo:
[asciicast](https://asciinema.org/a/627504)

Source is here: https://github.com/peterwilli/ToCase

**Pull request closed without merging**